tfx/components/example_gen/write_split.py

```diff
@@ -48,22 +52,38 @@ def process(self, e: Union[tf.train.Example, tf.train.SequenceExample,
 
 @beam.ptransform_fn
 @beam.typehints.with_input_types(
-    Union[tf.train.Example, tf.train.SequenceExample, bytes])
+    Union[tf.train.Example, tf.train.SequenceExample, bytes, Dict[str, Any]])
 @beam.typehints.with_output_types(beam.pvalue.PDone)
 def WriteSplit(
     example_split: beam.pvalue.PCollection,
     output_split_path: str,
     output_format: str,
+    exec_properties: Dict[str, Any]
 ) -> beam.pvalue.PDone:
-  """Shuffles and writes output split as serialized records in TFRecord."""
+  """
+  Shuffles and writes output split as serialized records in TFRecord or Parquet
+  """
   del output_format
 
+  output_payload_format = exec_properties.get(
+    standard_component_specs.OUTPUT_DATA_FORMAT_KEY)
+
+  if output_payload_format == example_gen_pb2.PayloadFormat.FORMAT_PARQUET:
+    # TODO: propagate schema.
+    schema = exec_properties.get("pyarrow_schema")
+    return (example_split
```

Comment:
To confirm, is this example_split contains PCollection of Dict or PCollection of bytes?

Reply:
in the case of parquet it contains PCollection of Dict